src/utils/src/Logger.c

```diff
@@ -38,15 +40,15 @@ VOID addLogMetadata(PCHAR buffer, UINT32 bufferLen, PCHAR fmt, UINT32 logLevel)
 #endif
 
     /* if something fails in getting time, still print the log, just without timestamp */
-    retStatus = generateTimestampStr(globalGetTime(), "%Y-%m-%d %H:%M:%S ", timeString, (UINT32) ARRAY_SIZE(timeString), &timeStrLen);
+    retStatus = generateTimestampStrInMilliseconds("%Y-%m-%d %H:%M:%S", timeString, (UINT32) ARRAY_SIZE(timeString), &timeStrLen);
```

Comment:
We still need a space here after the `%S`, right? 

So 
```
"%Y-%m-%d %H:%M:%S "
```

Instead of:

```
"%Y-%m-%d %H:%M:%S"
```

Reply:
Nope. We dont. The format for this function is: "%Y-%m-%d %H:%M:%S.ssssss ". So the space is added after the millisecond portion.

Reply:
I'm not following, according to the function comments for `generateTimestampStrInMilliseconds` the first parameter is 
```
 * @PCHAR   - IN - timestamp format string
```

The string passed in as the first argument is `%Y-%m-%d %H:%M:%S`, so that should be the timestamp format string.  But you're saying instead the format is:  `%Y-%m-%d %H:%M:%S.ssssss `?  I am confused, where is this latter format coming from?

Reply:
https://github.com/awslabs/amazon-kinesis-video-streams-pic/pull/187/files#diff-a54734ddfbb2b0634faf12eae8713f0162689aca80dbe80ee8f53b4a3799cee1R57-R63

The latter is appended inside `generateTimestampStrInMilliseconds` function.

Reply:
I've made comments elsewhere, it is not clear from the API definition that you actually do not expect the user to pass in a format string including milliseconds and that you manually append that part yourself.